packages/svelte/src/compiler/phases/2-analyze/css/css-prune.js

```diff
@@ -101,7 +109,37 @@ const visitors = {
 			}
 		}
 
-		if (
+		if (context.state.from_render_tag) {
+			// We're searching for a match that crosses a render tag boundary. That means we have to both traverse up
+			// the element tree (to see if we find an entry point) but also remove selectors from the end (assuming
+			// they are part of the render tag we don't see). We do all possible combinations of both until we find a match.
+			/** @type {Compiler.AST.RegularElement | Compiler.AST.SvelteElement | null} */
+			let element = context.state.element;
+
+			while (element) {
+				const selectors_to_check = selectors.slice();
```

Comment:
Wdym? We're traversing them back to front. That's what the rest of the pruning algorithm does.

Reply:
I mean a for loop with negative index rather copying the array and popping each item

Reply:
The result of pop isn't passed to anything, the resulting array that is one item less afterwards is. Looping with negative index wouldn't change the fact that we have to create a new array, if that's what you're concerned about (in fact we would have to create more since we'd have to do it for each iteration of the inner loop, rather of only each iteration of the outer loop)